Trellix c2c (#2635)
SumoLogic/sumologic-documentation: docs/send-data/hosted-collectors/cloud-to-cloud-integration-framework/trellix-mvisio-epo-source.md

---
id: trellix-mvisio-epo-source
title: Trellix mVision ePO Source
sidebar_label: Trellix mVision ePO
description: Learn how to collect event logs using the Trellix mVision ePO.
---
import useBaseUrl from '@docusaurus/useBaseUrl';

<img src={useBaseUrl('img/send-data/trellix-logo.png')} alt="trellix-logo" width="70" />

> Trellix is a cybersecurity company that provides cloud-based security solutions for cybersecurity attacks. It provides hardware, software, and services to investigate cybersecurity attacks, protect against malicious software, and analyze IT security risks.

Reviewer: ```suggestion
Trellix is a cybersecurity company that provides cloud-based security solutions for cybersecurity attacks. It provides hardware, software, and services to investigate cybersecurity attacks, protects against malicious software, and analyzes IT security risks.
```

> mVision ePO is a key component of the Trellix security management platform, which provides unified management of endpoint, network, and data security, which reduces incident response time, strengthen protection, simplify risk and security management with automation, and end-to-end network visibility and security.

Reviewer: ```suggestion
mVision ePO is a key component of the Trellix security management platform, which provides unified management of endpoint, network, and data security. This can reduce incident response time, strengthen protection, simplify and automate risk and security management, and provide end-to-end network visibility and security.
```

> mVision ePO is a key component of the Trellix Security Management Platform, which provides Unified Management of Endpoint, Network, and Data Security, which reduces Incident Response Times, Strengthen Protection, Simplify Risk and Security Management with Automation, and End-to-end Network Visibility and Security.

Reviewer 2: ```suggestion
mVision ePO is a key component of the Trellix Security Management Platform, which provides unified management of endpoint, network, and data security, reduces incident response times, strengthens protection, simplify risk and security management with Automation, and end-to-end network visibility and security.
```

Author: No, this needs a fix. I will update.

## Data Source

The Trellix integration fetches event logs using the [Trellix mVision ePO API](https://developer.manage.trellix.com/mvision/apis/v2-events).

## Set up and configuration

> The Trellix mVision ePO source requires you to provide Client ID, Client Secret, and API key. To obtain these, follow the steps below:

Reviewer: ```suggestion
The Trellix mVision ePO source requires you to provide a Client ID, Client Secret, and API key. To obtain these, follow the steps below.
```

> 1. Login to the [Trellix Developer Portal](https://developer.manage.trellix.com/).

Reviewer 2: ```suggestion
1. Log in to the [Trellix Developer Portal](https://developer.manage.trellix.com/).
```

> 1. Go to **Self service > [API Access Management](https://developer.manage.trellix.com/mvision/selfservice/access_manag)**. <br/><img src={useBaseUrl('img/send-data/self-service.png')} alt="self-service.png" style={{border: '1px solid black'}} width="600" />

Reviewer 2: ```suggestion
1. Go to **Self service** > **[API Access Management](https://developer.manage.trellix.com/mvision/selfservice/access_manag)**. <br/><img src={useBaseUrl('img/send-data/self-service.png')} alt="self-service.png" style={{border: '1px solid black'}} width="600" />
```

Reviewer 2: `https://developer.manage.trellix.com/mvision/selfservice/access_manag`

This looks like an incomplete URL.

Reviewer 2: It's good to write an action button name after any actionable items. Like **Self Service** menu or tab or whatever it is.

Author: That is the complete URL, the page works with this URL.

> 1. Go to **Self service** menu, and click **[API Access Management](https://developer.manage.trellix.com/mvision/selfservice/access_manag)**. <br/><img src={useBaseUrl('img/send-data/self-service.png')} alt="self-service.png" style={{border: '1px solid black'}} width="600" />

Reviewer: ```suggestion
1. Go to the **Self service** menu and click [**API Access Management**](https://developer.manage.trellix.com/mvision/selfservice/access_manag).<br/><img src={useBaseUrl('img/send-data/self-service.png')} alt="self-service.png" style={{border: '1px solid black'}} width="600" />
```

> 1. Go to the **Self service** menu and click [**API Access Management**](https://developer.manage.trellix.com/mvision/selfservice/access_manag).<br/><img src={useBaseUrl('img/send-data/self-service.png')} alt="self-service.png" style={{border: '1px solid black'}} width="600" />

Reviewer 2: ```suggestion
1. Go to the **Self service** menu, and click [**API Access Management**](https://developer.manage.trellix.com/mvision/selfservice/access_manag).<br/><img src={useBaseUrl('img/send-data/self-service.png')} alt="self-service.png" style={{border: '1px solid black'}} width="600" />
```

Reviewer 2: Go to the **Self service** menu, and click [**API Access Management**] tab. 

Can you write the action button name there? I think it's a tab, check plz.

Author: **API Access Management** is an option in Self service menu. So I will call it as option.

>     1. Copy the **API key** from the API Access Management section. <br/><img src={useBaseUrl('img/send-data/api-key.png')} alt="api-key" style={{border: '1px solid black'}} width="600" />

Reviewer 2: ```suggestion
    1. Copy **API key** from the API Access Management section. <br/><img src={useBaseUrl('img/send-data/api-key.png')} alt="api-key" style={{border: '1px solid black'}} width="600" />
```

    1. In the **Credential Configuration** section, select **Events** as the scope in **APIs** for a given **Client Type**.

>     1. Click **Request**, for IAM Client type approval.<br/><img src={useBaseUrl('img/send-data/credential-configuration.png')} alt="credential-configuration" style={{border: '1px solid black'}} width="600" />

Reviewer 2: ```suggestion
    1. Click **Request** for IAM Client type approval.<br/><img src={useBaseUrl('img/send-data/credential-configuration.png')} alt="credential-configuration" style={{border: '1px solid black'}} width="600" />
```

>     1. Once your IAM Client type is approved, generate **Client ID** and **Client Secret**. <br/><img src={useBaseUrl('img/send-data/generate.png')} alt="generate" style={{border: '1px solid black'}} width="600" />

Reviewer: ```suggestion
    1. Once your IAM Client type is approved, generate the **Client ID** and **Client Secret**. <br/><img src={useBaseUrl('img/send-data/generate.png')} alt="generate" style={{border: '1px solid black'}} width="600" />
```

## States

The Trellix mVision ePO source is a security platform that provides cloud-based security solutions for cybersecurity attacks.

> 

Reviewer 2: ```suggestion
```

> When you create an Trellix mVision ePO source, it goes through the following stages:

Reviewer 2: ```suggestion
When you create a Trellix mVision ePO source, it goes through the following stages.
```

1. **Pending**. Once the Source is submitted, it is validated, stored, and placed in a **Pending** state.
1. **Started**. A collection task is created on the Hosted Collector.
1. **Initialized**. The task configuration is complete in Sumo Logic.
1. **Authenticated**. The Source successfully authenticated with Trellix mVision ePO.
1. **Collecting**. The Source is actively collecting data from Trellix mVision ePO.

If the Source has any issues during any one of these states, it is placed in an **Error** state.

> 

Reviewer 2: ```suggestion
```

1. (Optional) For **Source Category**, enter any string to tag the output collected from the Source. Category metadata is stored in a searchable field called `_sourceCategory`.
1. (Optional) **Fields**. Click the **+Add** button to define the fields you want to associate. Each field needs a name (key) and value.
   * ![green check circle.png](/img/reuse/green-check-circle.png) A green circle with a check mark is shown when the field exists in the Fields table schema.
   * ![orange exclamation point.png](/img/reuse/orange-exclamation-point.png) An orange triangle with an exclamation point is shown when the field doesn't exist in the Fields table schema. In this case, an option to automatically add the nonexistent fields to the Fields table schema is provided. If a field is sent to Sumo Logic that does not exist in the Fields schema it is ignored, known as dropped.
1. Enter the **Client ID** of your Trellix platform.
1. Enter the **Client Secret** of your Trellix platform.
1. Enter the **API Key** for authorization collected from the Trellix platform.

> 1. (Optional) The **Polling Interval** is set for 5 minutes by default, you can adjust it based on your needs. This sets how often the Source checks for new data.

Reviewer: ```suggestion
1. (Optional) The **Polling Interval** is set for 5 minutes by default. You can adjust it based on your needs. This sets how often the Source checks for new data.
```

1. When you are finished configuring the Source, click **Save**.

### Error Types

> When Sumo Logic detects an issue it is tracked by Health Events. The following table shows the three possible error types, the reason for the error, if the source attempts to retry, and the name of the event log in the Health Event Index.

Reviewer 2: ```suggestion
When Sumo Logic detects an issue, it is tracked by Health Events. The following table shows the three possible error types, the reason for the error, if the source attempts to retry, and the name of the event log in the Health Event Index.
```

| `sourceType` | String | Yes | Use `Universal` for Trellix mVision ePO. | not modifiable |

### Config Parameters

| Parameter | Type | Required | Description | Access |
|:---|:---|:---|:---|:---|
| `name` | String | Yes | Type the desired name of the Source and it must be unique per Collector. This value is assigned to the `metadata field _source`.  | modifiable |
| `description` | String  | No | Type the description of the Source. | modifiable |
| `category` | String | No | Type the category of the source. This value is assigned to the metadata field `_sourceCategory`. | modifiable |
| `fields` | JSON Object | No | JSON map of key-value fields (metadata) to apply to the Collector or Source. Use the boolean field `_siemForward` to enable forwarding to SIEM. | modifiable |
| `apiKey` | String | Yes | API key used for Authorization.  | modifiable |
| `clientID` | String | Yes | Client ID of your account. | modifiable |
| `clientSecret` | String | Yes | Client Secret of your account. | modifiable |
| `pollingInterval` | Integer | No | Choose how often the Source checks for new data (In minutes). | modifiable |

### JSON Example

```json

> {
>   "api.version": "v1",
>   "source": {
>     "config": {
> 	"name": "Trellix",
>   	"clientID": "xxxxxxxxxxxxxx",
> 	"clientSecret": "******",
>  	"apiKey": "*******",
> "pollingInterval": 5
>     	},
>     "schemaRef": {
>       "type": "Trellix mVision ePO"
>     },
>     "sourceType": "Universal"
>   }
> }

Reviewer: indentation

```suggestion
{
  "api.version":"v1",
  "source":{
    "config":{
      "name":"Trellix",
      "clientID":"xxxxxxxxxxxxxx",
      "clientSecret":"******",
      "apiKey":"*******",
      "pollingInterval":5
    },
    "schemaRef":{
      "type":"Trellix mVision ePO"
    },
    "sourceType":"Universal"
  }
}
```
